src/adapter/src/catalog.rs

```diff
@@ -7452,6 +7453,18 @@ impl SessionCatalog for ConnCatalog<'_> {
         }
     }
 
+    fn get_privileges(&self, id: &ObjectId) -> Option<&PrivilegeMap> {
+        match id {
+            ObjectId::Cluster(id) => Some(self.get_cluster(*id).privileges()),
+            ObjectId::Database(id) => Some(self.get_database(id).privileges()),
+            ObjectId::Schema((database_spec, schema_spec)) => {
+                Some(self.get_schema(database_spec, schema_spec).privileges())
+            }
+            ObjectId::Item(id) => Some(self.get_item(id).privileges()),
+            ObjectId::ClusterReplica(_) | ObjectId::Role(_) => None,
```

Comment:
Intentional that this is doing nothing yet? It's also not covered. via https://buildkite.com/materialize/coverage/builds/82

Reply:
Yes, replicas and role's don't have privileges I've added some rust tests below for this and others.